src/main/java/net/coderbot/iris/shaderpack/PackDirectives.java

```diff
@@ -20,40 +20,37 @@
 	private boolean separateAo;
 	private boolean oldLighting;
 	private boolean particlesBeforeDeferred;
-	private OptionalBoolean shadowCulling;
 	private Object2ObjectMap<String, Object2BooleanMap<String>> explicitFlips = new Object2ObjectOpenHashMap<>();
 
 	private final PackRenderTargetDirectives renderTargetDirectives;
 	private final PackShadowDirectives shadowDirectives;
 
-	private PackDirectives(Set<Integer> supportedRenderTargets) {
+	private PackDirectives(Set<Integer> supportedRenderTargets, PackShadowDirectives packShadowDirectives) {
 		noiseTextureResolution = 256;
 		sunPathRotation = 0.0F;
 		ambientOcclusionLevel = 1.0F;
 		renderTargetDirectives = new PackRenderTargetDirectives(supportedRenderTargets);
-		shadowDirectives = new PackShadowDirectives();
+		shadowDirectives = packShadowDirectives;
 	}
 
 	PackDirectives(Set<Integer> supportedRenderTargets, ShaderProperties properties) {
-		this(supportedRenderTargets);
+		this(supportedRenderTargets, new PackShadowDirectives(properties.getShadowTerrain().orElse(true), properties.getShadowTranslucent().orElse(true), properties.getShadowEntities().orElse(true), properties.getShadowBlockEntities().orElse(true), properties.getShadowCulling()));
 		areCloudsEnabled = properties.areCloudsEnabled();
 		underwaterOverlay = properties.getUnderwaterOverlay().orElse(false);
 		vignette = properties.getVignette().orElse(false);
 		rainDepth = properties.getRainDepth().orElse(false);
 		separateAo = properties.getSeparateAo().orElse(false);
 		oldLighting = properties.getOldLighting().orElse(false);
 		explicitFlips = properties.getExplicitFlips();
-		shadowCulling = properties.getShadowCulling();
 		particlesBeforeDeferred = properties.getParticlesBeforeDeferred().orElse(false);
 	}
 
 	PackDirectives(Set<Integer> supportedRenderTargets, PackDirectives directives) {
-		this(supportedRenderTargets);
+		this(supportedRenderTargets, new PackShadowDirectives(directives.getShadowDirectives().shouldRenderTerrain(), directives.getShadowDirectives().shouldRenderTranslucent(), directives.getShadowDirectives().shouldRenderEntities(), directives.getShadowDirectives().shouldRenderBlockEntities(), directives.getShadowDirectives().getCullingState()));
```

Comment:
Why not make ShadowDirectives have a "copy constructor" here or just pass it as-is?